dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/DependencyServiceImpl.java

```diff
@@ -65,7 +74,40 @@ private void initMaven() throws ClassNotFoundException, NoSuchMethodException, M
             URL[] mavenJarUrls = getUrls(libDir);
 
             mavenClassLoader = new URLClassLoader(mavenJarUrls, parentClassLoader);
+            MAVEN_EXECUTE_METHOD = Class.forName(MAVEN_LAUNCHER_CLASS_NAME, true, mavenClassLoader).
+                    getMethod(MAVEN_LANUCHER_METHOD_NAME, String[].class);
+            initMavenLogs();
+        }
+    }
+
+    private void initMavenLogs() {
+        String logFolderPath = calculateLogFolderPath();
+        File logFolderFile = new File(logFolderPath);
+        logFolderFile.mkdirs();
+        File mavenLogFolderFile = new File(logFolderFile, "maven");
+        mavenLogFolderFile.mkdirs();
+        this.mavenLogFolder = mavenLogFolderFile.getAbsolutePath();
+    }
+
+    private String calculateLogFolderPath() {
+        Enumeration e = Logger.getRootLogger().getAllAppenders();
+        while ( e.hasMoreElements() ){
+            Appender app = (Appender)e.nextElement();
+            if ( app instanceof FileAppender){
+                String logFile = ((FileAppender) app).getFile();
+                return new File(logFile).getParentFile().getAbsolutePath();
+            }
         }
+        return new File(System.getProperty("app.home"), "logs").getAbsolutePath();

```

Comment:
IMHO it's worth extracting these strings as constants: app.home, logs, maven etc..